src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,372 @@
+        type=str,
+        nargs='*',
+        help="The names of the tensors that are to be interpolated. "
+        "Accepts regular expression. "
+        "If none are specified, all tensors are interpolated.")
+
+    parser.add_argument("--grid_extents",
+                        type=int,
+                        nargs='+',
+                        default=3,
+                        help="The extents of the target grid. "
+                        "Can be different for each dimension e.g. '3 5 4'")
+
+    parser.add_argument("--grid_basis",
+                        type=str,
+                        nargs='+',
+                        choices=basis_dict.keys(),
+                        default="Legendre",
+                        help="The basis of the target grid. "
+                        "Can be different for each dimension")
+
+    parser.add_argument("--grid_quadrature",
+                        type=str,
+                        nargs='+',
+                        choices=quadrature_dict.keys(),
+                        default="GaussLobatto",
+                        help="The quadrature of the target grid. "
+                        "Can be different for each dimension")
+
+    parser.add_argument("--source_volume",
+                        type=str,
+                        nargs='+',
+                        default=["/VolumeData"],
+                        help="The name of the volume data object within the "
+                        "source file in which the data is contained")
+
+    parser.add_argument(
+        "--target_volume",
+        type=str,
+        nargs='+',
+        default=["/VolumeDataInterpolated"],
+        help="The name of the volume data object within the target "
+        "file where the data is written.")
+
+    parser.add_argument(
+        "--observation_start",
+        type=float,
+        default=-np.inf,
+        help="Disregard all observations with value before this point")
+
+    parser.add_argument(
+        "--observation_end",
+        type=float,
+        default=np.inf,
+        help="Disregard all observations with value after this point")
+
+    parser.add_argument(
+        "--observation_stride",
+        type=float,
+        default=1,
+        help="Stride through observations with this step size.")
+
+    parser.add_argument(
+        "-j",
+        "--jobs",
+        type=int,
+        default=1,
+        help="The maximum number of processes to be started. "
+        "A process is spawned for each source file up to this number.")
+
+    parser.add_argument('-v',
+                        '--verbose',
+                        action='count',
+                        default=0,
+                        help="Verbosity (-v, -vv, ...)")
+
+    return parser.parse_args(sys_args)
+
+
+if __name__ == "__main__":
+
+    parsed_args = _parse_args(sys.argv[1:])
+
+    logging.basicConfig(level=logging.WARNING - parsed_args.verbose * 10)
+
+    # check if wildcards in source_files
```

Comment:
I think many of these checks won't be necessary anymore with the simplifications to the CLI I suggested above. I'll hold off reviewing them in detail until we have decided on the CLI.